sdk/core/AzureCore/Source/Pipeline/Transport/HTTPHeader.swift

```diff
@@ -159,6 +159,13 @@ extension HTTPHeaders {
     /// - Parameters:
     ///   - values: A dictionary of `HTTPHeader` values to populate the `HTTPHeaders` instance.
     public init(_ values: [HTTPHeader: String]) {
+        self.init(minimumCapacity: values.underestimatedCount)
+        for (key, value) in values {
+            self[key.rawValue] = value
+        }
+    }
+
+    public init(_ values: [String: String]) {
```

Comment:
I think we intentionally avoided having such an initializer in order to ensure that any HTTP headers that needed to accompany a request would be added to the appropriate structures (`HTTPHeader`, `StorageHTTPHeader`, etc)

Reply:
The problem is that the existing initializer will not work with mixed types: `[HTTPHeader, StorageHTTPHeader]`, even though both of these are simply type aliases for string-based dictionaries! Swift will complain that `StorageHTTPHeader` is not an `HTTPHeader`. Perhaps an enum is not the best structure to use here?

Reply:
Yeah, we have initializers for both types, but as you point out it doesn't work if they're mixed. Thankfully it's easy enough to create it with one type and then add members of another, IIRC only the initializers suffer from this problem. I'm not sure if it could be fixed with the appropriate use of generics or not, but maybe worth investigating. That said, it's not really consumer-facing (more a problem for library authors) so I'm less motivated to spend the cycles to make this more elegant.

Reply:
Makes sense. I reverted the second initializer. Should be good now.